ui/src/app/approval-list/approval-list.component.spec.ts

```diff
@@ -1,25 +1,91 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
 
-import { ApprovalListComponent } from './approval-list.component';
+import {ApprovalListComponent} from './approval-list.component';
+import {MatTableModule} from '@angular/material';
+import {of} from 'rxjs';
+import {MatDialog} from '@angular/material/dialog';
+import {FilterInputComponent} from '../filter-input/filter-input.component';
+import {CustomMaterialModule} from '../core/material.module';
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {CdkTableModule} from '@angular/cdk/table';
+import {HttpClient, HttpErrorResponse} from '@angular/common/http';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {LoadingService} from '@services/loading.service';
+import {LoginService} from '../auth/login.service';
+import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
+import {MatSnackBar} from '@angular/material/snack-bar';
+
+class MatDialogMock {
+  public open() {
+    return {afterClosed: () => of({confirmed: true})};
+  }
+}
+
+class MatSnackBarStub {
+  open() {
+    return {
+      onAction: () => of({})
+    };
+  }
+}
 
 describe('ApprovalListComponent', () => {
   let component: ApprovalListComponent;
   let fixture: ComponentFixture<ApprovalListComponent>;
+  let httpClient: HttpClient;
+  let httpTestingController: HttpTestingController;
+  let matSnackBar: MatSnackBar;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ ApprovalListComponent ]
+      imports: [
+        CustomMaterialModule,
+        FormsModule,
+        ReactiveFormsModule,
+        MatTableModule,
+        CdkTableModule,
+        HttpClientTestingModule,
+        BrowserAnimationsModule
+      ],
+      declarations: [ApprovalListComponent, FilterInputComponent],
+      providers: [
+        {provide: MatDialog, useClass: MatDialogMock},
+        {provide: LoadingService, useValue: {setLoading: () => null}},
+        {provide: LoginService, useValue: {}},
+        {provide: MatSnackBar, useClass: MatSnackBarStub}
+      ]
     })
-    .compileComponents();
+      .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(ApprovalListComponent);
     component = fixture.componentInstance;
+    matSnackBar = TestBed.get(MatSnackBar);
+    httpTestingController = TestBed.get(HttpTestingController);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should notify users of successful approval', () => {
```

Comment:
@tommckennon I reworked the tests for the approval list... plenty more could be used but I think these are good examples of testing certain things.  basically... does confirmApproval trigger the snackbar to return a certain message based on if it gets a good or error response from http client.